directory/cli/src/user.py

```diff
@@ -303,8 +300,10 @@ def _transform_options(argument, options):
 def _transform_create_options(argument, options):
     _validate_blacklist_users(argument)
     _validate_create_uid(options['uid'])
-    if options['gidnumber'] == None and utils.detect_user_config():
-        options['gidnumber'] = utils.get_user_config('DEFAULT_GID')
+
+    group_id = _get_group_id('clusterusers')
```

Comment:
In your PR description you say that the precedence for which GID should be set by default is:

> If this value does not exist it then looks for the clusterusers group and assigns accordingly. However if both of these are unavailable IPA should assign the primary group to its own default.

however as far as I can see, if the `clusterusers` group is unavailable, this function call will error and we'll crash out without creating the user (see https://github.com/alces-software/userware/pull/73/files#diff-61afed74b6f67d8049810ac2b061ad1bR495). I could be mistaken, and not sure if this is what we want to happen (as if the `clusterusers` group, which I think is supposed to always exist, doesn't exist then that sounds like it could be a sign of a bigger issue), but thought I'd point this out. If this is correct, then we should consider what we do want to happen in this situation - @ColonelPanicks may have thoughts.

Also, regardless of whether this both works as described or not, and is what we actually want to happen or not, it would be worth adding test coverage for the different situations - atm I think we have enough implicit coverage that the `clusterusers` GID is normally used in all the other tests, but could be worth adding a test that the value of `DEFAULT_GID` takes precedence, when this is present in the config file, and that whatever we decide we want to happen when `DEFAULT_GID` is not set and the `clusterusers` group does not exist, does actually happen.